actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/util/OTPUtil.java

```diff
@@ -52,6 +53,23 @@ public static String generateOTP() {
     return String.valueOf(code);
   }
 
+  /**
+   * if otp length is less than 4 , regenerate otp (max retry 3)
+   *
+   * @param otp
+   * @return
+   */
+  public static String checkOtpCodeLength(String otp) {
```

Comment:
This is not just checking but also generating. The practice is to call these kind of methods as 'ensure' - once a caller calls this, the caller can be safely assuming that the job has been done. Please rename this to 'ensureOtpLength'

Also, add a log line before otp with the noOfAttempts. Even after the retry count limit is exceed, it may be that the OTP is still not 4 in length. Consider multiplying the otp by 10000 as a last resort to these.

Reply:
done